packages/jest-cli/src/Runtime/Runtime.js

```diff
@@ -401,19 +444,20 @@ class Runtime {
     }
 
     const delimiter = path.delimiter;
-    const id = moduleType + delimiter + absolutePath + delimiter + mockPath;
+    const id = moduleType + delimiter + (absolutePath || '') +
+      delimiter + (mockPath || '');
     normalizedIDCache[key] = id;
     return id;
   }
 
-  _getVirtualMockPath(from, moduleName) {
+  _getVirtualMockPath(from: string, moduleName: string) {
     if (moduleName[0] !== '.' && moduleName[0] !== '/') {
       return moduleName;
     }
     return path.normalize(path.dirname(from) + '/' + moduleName);
   }
 
-  _shouldMock(from, moduleName) {
+  _shouldMock(from: string, moduleName: string) {
```

Comment:
For `from` can we use `Path` from the `types/Config` module? It maps to string but makes the code more readable.